### PR TITLE
 Add gRPC version of LoadMap().

### PR DESCRIPTION
**Pull request by pifon2a**
Later they will be used in map_builder_stub and load_map_handler.

### CODE REVIEW

File: cartographer_grpc/io/inmemory_proto_stream.cc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer_grpc/io/inmemory_proto_stream.h"
+
+namespace cartographer_grpc {
+namespace io {
+
+bool InmemoryProtoStreamReader::ReadProto(google::protobuf::Message* proto) {
```
Comment (cschuet) [inline cartographer_grpc/io/inmemory_proto_stream.cc]: Rename to  InMemoryProtoStreamReader

File: cartographer_grpc/io/inmemory_proto_stream_test.cc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+namespace cartographer_grpc {
+namespace io {
+namespace {
+
+using ::cartographer::mapping::proto::PoseGraph;
+using ::cartographer::mapping::proto::SerializedData;
+
+// Test if a queue that contains PoseGraph and then SerializedData can be read.
+TEST(InmemoryProtoStreamTest, ReadCorrectQueue) {
```
Comment (cschuet) [inline cartographer_grpc/io/inmemory_proto_stream_test.cc]: InmemoryProtoStreamTest => InMemoryProtoStreamTest

File: cartographer_grpc/io/inmemory_proto_stream.cc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  if (eof()) return false;
+
+  switch (load_map_requests_.front().map_chunk_case()) {
```
Comment (cschuet) [inline cartographer_grpc/io/inmemory_proto_stream.cc]: Let's make this more general and not talk about the concrete types here. Use a queue<unqiue_ptr<Message>>.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 2.

**Pull request merged**